architecture/build-service.md

```diff
@@ -19,170 +19,114 @@ The Build Service is dependent on the following services:
+1. Removes `build.appstudio.openshift.io/request` annotation.
+
+PaC unprovision:
+1. Removes finalizer `pac.component.appstudio.openshift.io/finalizer`.
+1. Removes webhook from repository if GitHub App isn't used and the repository isn't used in another component.
+1. Creates merge request in the user code repository removing PipelineRun definitions.
+1. Sets `build.appstudio.openshift.io/status` annotation with either error, or state `disabled` and merge request link.
+1. Removes `build.appstudio.openshift.io/request` annotation.
+
+Trigger PaC build:
+1. Triggers push pipeline via PaC incoming webhook, requires pipeline run name to be the same as it was named during provisioning `$COMPONENT_NAME-on-push`.
+1. Sets `build.appstudio.openshift.io/status` annotation when error occures.
+1. Removes `build.appstudio.openshift.io/request` annotation.
 
-Component Build Controller is working in two modes:
-- Default Mode
-  - Mode where PipelineRuns are created **directly** by the Build Service.
-- Custom Mode
-  - Integrates with Pipelines As Code, the Component Build Controller creates PipelineRuns definitions in the user code repository.
-  - The prerequisite is to have installed GitHub App which is used by the Build Service in the user's repository.
-
-Default mode:
-1. Wait until the component has `.status.devfile` set
-1. Get data from `.status.devfile` and use them for creating PipelineRun
-1. Set annotation on the component - `appstudio.openshift.io/component-initial-build: processed`
-
-Delete annotation `appstudio.openshift.io/component-initial-build` on Component to retrigger PipelineRun in Default mode.
-
-For the selection of Custom mode the Component CR must have set annotation `appstudio.openshift.io/pac-provision` to `request`.
+#### PipelineRun selection
+Available and default pipelines are in the config map present on the cluster in controller's namespace
+[build pipelines config](https://github.com/redhat-appstudio/infra-deployments/blob/main/components/build-service/base/build-pipeline-config/build-pipeline-config.yaml).
 
-Custom Mode:
-1. Wait until the component has `.status.devfile` set
-1. Get data from `.status.devfile`, use them for creating `.tekton` folder in the user's repository and create Repository CR so that Pipelines as Code is configured to monitor the user's repository.
-1. Change the value of `appstudio.openshift.io/pac-provision` annotation to `done` in case the previous step was successful, in case of an issue the annotation is set to `error` and annotation `appstudio.openshift.io/pac-provision-error` with more details about the error is added.
+Build pipeline is selected based on `build.appstudio.openshift.io/pipeline` annotation,
+when annotation is missing, annotation with default pipeline (based on config map) will be added.
 
-#### PipelineRun selection
+Annotation value is json in string eg. `'{"name":"docker-build","bundle":"latest" }`.
```

Comment:
Extra space at the end of annotation value

Reply:
removed